InTheMiddle2/InTheMiddle2/ViewController.swift

```diff
@@ -136,3 +188,42 @@ class ViewController: UIViewController, GMSMapViewDelegate, CLLocationManagerDel
     }
 }
 
+//GMSAutocompleteDelegate, for autocomplete location search
+extension ViewController: GMSAutocompleteViewControllerDelegate {
+    
+    func viewController(_ viewController: GMSAutocompleteViewController, didFailAutocompleteWithError error: Error) {
+        print("Error: \(error)")
+    }
+    
+    func viewController(_ viewController: GMSAutocompleteViewController, didAutocompleteWith place: GMSPlace) {
+        //change map location
+        let camera = GMSCameraPosition.camera(withLatitude: place.coordinate.latitude, longitude: place.coordinate.longitude, zoom: 10.25)
+        
+        if locationSelected == .locationA {
+            locationA?.text = "\(place.coordinate.latitude), \(place.coordinate.longitude)"
+            locationStart = CLLocation(latitude: place.coordinate.latitude, longitude: place.coordinate.longitude)
+            //createLocationMarker()
+        } else {
+            locationB?.text = "\(place.coordinate.latitude), \(place.coordinate.longitude)"
+            locationEnd = CLLocation(latitude: place.coordinate.latitude, longitude: place.coordinate.longitude)
+            //createLocationMarker()
+        }
+        
+        self.googleMapsView?.camera = camera
+//       self.googleMapsView?.animate(to: camera)
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    func wasCancelled(_ viewController: GMSAutocompleteViewController) {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    func didRequestAutocompletePredictions(_ viewController: GMSAutocompleteViewController) {
+        UIApplication.shared.isNetworkActivityIndicatorVisible = true
+    }
+    
+    func didUpdateAutocompletePredictions(_ viewController: GMSAutocompleteViewController) {
```

Comment:
I think there is a way to display the updated markers, You may need to recall viewdidload or call a function that will reload the updated markers.